undertow/src/main/java/org/wildfly/extension/undertow/ApplicationSecurityDomainDefinition.java

```diff
@@ -335,8 +335,13 @@ protected RemoveHandler(AbstractAddStepHandler addOperation) {
         protected void performRemove(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
             if (context.isResourceServiceRestartAllowed()) {
                 String securityDomainName = context.getCurrentAddressValue();
-                context.removeService(new SingleSignOnManagerServiceNameProvider(securityDomainName).getServiceName());
-                context.removeService(new SingleSignOnSessionFactoryServiceNameProvider(securityDomainName).getServiceName());
+                context.addStep(new OperationStepHandler() {
+                    @Override
+                    public void execute(OperationContext operationContext, ModelNode modelNode) throws OperationFailedException {
+                        context.removeService(new SingleSignOnManagerServiceNameProvider(securityDomainName).getServiceName());
+                        context.removeService(new SingleSignOnSessionFactoryServiceNameProvider(securityDomainName).getServiceName());
+                    }
+                }, OperationContext.Stage.RUNTIME);
```

Comment:
@pferraro I am going to merge it since I am on phone and won't be in work mode by the time I get to a keyboard